src/hotspot/share/runtime/os.hpp

```diff
@@ -25,7 +25,7 @@
 #ifndef SHARE_RUNTIME_OS_HPP
 #define SHARE_RUNTIME_OS_HPP
 
-#include "jvm.h"
+#include "jvm_md.h"
```

Comment:
jvm_md.h should not be included directly by any file - it is an internal implementation detail.

Reply:
os.hpp is used by almost all HotSpot .o files, so if we keep jvm.h here, that defeats the purpose of this PR. As David and I discussed offline, jvm_md.h is a kludgy legacy porting layer for stuff like DIR, which is used by os.hpp.

jvm_md.h contains comments like the following that haven't been touched for decades .... so it's probably a good idea to not disturb it now.

```
/*
 * This file is currently collecting system-specific dregs for the
 * JNI conversion, which should be sorted out later.
 */
```

Reply:
What does os.hpp need from jvm_md.h ? Can that be moved to os.cpp ? I guess I can look at the file and answer my own question.

Reply:
One reason is that os.hpp uses `DIR` in function declarations, and `DIR` is declared in jvm_md.h in a platform-dependent way. So we can't forward-declare `DIR` in os.hpp.